### PR TITLE
Add target object to Sighting entity

### PR DESCRIPTION
**Pull request by msprunck**
Connected to threatgrid/iroh#518

### CODE REVIEW

**Comment by craigbro** [comment]
Thanks for putting this together, it captures most of the information we need, and shows the most direct encoding of it.

We should be using existing CTIM patterns where possible.  So target should have a set of observables which define, or "triangulate" it.  So, encode the IPs, guid, and mac addr as observables.  We also need to handle targets that may have multiple IP interfaces.

I also think we should be using the taxonomy of devices that we use in sighting.sensor field to identify the target.  Some possible target types are servers, specific web services, email accounts, network devices (routers,firewalls), mobile devices (endpoint still?)

/code
:target {:type "endpoint"
              :observables
                {:type "hostname" :value "Demo_Cta"}
                {:type "amp_computer_guid" :value "68e94bf7-e239-4821-90d6-b7eaa0233443"}
                {:type "ip" :value "100.213.110.122"}
                {:type "ip" :value "136.184.130.98"}
                {:type "mac_address" :value "85:28:cb:6a:21:41"}}
              :properties_data_table
                {.. data table object encoding operating system, version etc... }}

**Comment by msprunck** [comment]
The distinction between internal and external IP address is not important ?

File: src/ctim/schemas/vocabularies.cljc [hunk 1]
```diff
@@ -335,7 +335,12 @@
     "email"
     "imei"
     "imsi"
-    "amp-device"})
+    "amp-device"
```
Comment (yogsototh) [inline src/ctim/schemas/vocabularies.cljc]: using a dash is not coherent with the new observable types that use an underline char.

Comment (msprunck) [inline src/ctim/schemas/vocabularies.cljc]: I use the underscore for new values only. For backward compatibility, I kept the dash for old values.

The dash character is not valid for GraphQL enums

**Pull request merged**